test/integration/scheduler_perf/util.go

```diff
@@ -166,55 +166,55 @@ func (*metricsCollector) run(ctx context.Context) {
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
 	for _, metric := range pc.Metrics {
-		dataItem := collectHistogram(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
-		}
+		itemCollection := collectHistogram(metric, pc.labels)
+		dataItems = append(dataItems, itemCollection...)
 	}
 	return dataItems
 }
 
-func collectHistogram(metric string, labels map[string]string) *DataItem {
-	hist, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric)
+func collectHistogram(metric string, labels map[string]string) []DataItem {
+	var dataItems []DataItem
+	histMap, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric)
 	if err != nil {
 		klog.Error(err)
 		return nil
 	}
-	if hist.Histogram == nil {
-		klog.Errorf("metric %q is not a Histogram metric", metric)
-		return nil
-	}
-	if err := hist.Validate(); err != nil {
-		klog.Error(err)
-		return nil
-	}
+	for label, hist := range histMap {
+		if err := hist.Validate(); err != nil {
+			klog.Error(err)
+			return nil
+		}
 
-	q50 := hist.Quantile(0.50)
-	q90 := hist.Quantile(0.90)
-	q99 := hist.Quantile(0.95)
-	avg := hist.Average()
+		q50 := hist.Quantile(0.50)
```

Comment:
These lines are may also be causing golint to fail.

Reply:
Done

Reply:
there is no issue in this line, golint error is fixed.